src/main/java/nl/tudelft/contextproject/script/Script.java

```diff
@@ -213,14 +244,24 @@ public int hashCode() {
      */
     @Override
     public Shot next() {
-        if (current > -1) {
-            Shot old = shots.get(current);
-            timelines.get(old.getCamera().getNumber()).nextPreset(old);
-        }
+        updateOldCamCaller();
 
         current++;
         Shot next = shots.get(current);
         next.execute();
         return next;
     }
+    
+    public class UpdateTask extends TimerTask {
```

Comment:
Different classes should be in different files. Please create a new file for this class.